esmvalcore/preprocessor/_regrid.py

```diff
@@ -185,8 +189,43 @@ def _attempt_irregular_regridding(cube, scheme):
     return False
 
 
+def _add_regridded_aux_coords(cube, regridded_cube, target_grid):
+    """Get regridded :class:`iris.coords.AuxCoord` and corresponding dims."""
+    if regridded_cube.ndim != cube.ndim:
+        return
+    for coord in cube.coords(dim_coords=False):
+        if regridded_cube.coords(coord.name()):
+            continue
+        if coord.bounds is not None:
+            logger.warning(
+                "Auxiliary coordinate '%s' cannot be regridded since it "
+                "contains bounds", coord.name())
+            continue
+        logger.debug(
+            "Regridding (bound-less) auxiliary coordinate '%s' which could "
+            "not be regridded by iris", coord.name())
+
+        # Broadcast to shape of whole cube
+        coord_dims = cube.coord_dims(coord)
+        points = iris.util.broadcast_to_shape(coord.core_points(),
+                                              cube.shape,
+                                              coord_dims)
```

Comment:
Do I understand correctly that this has the potential to incur a hefty additional cost? A 1d aux coord, even a scalar coord might be blown into a long timeseries of 3d data, right? If I understand [iris](https://github.com/SciTools/iris/blob/06e9e2b5eae6e903e486ce12162c242d25ee4e5d/lib/iris/analysis/_regrid.py#L1003) correctly, this is not necessary. Source and target only need to share x and y dim. What happens without the broadcast?